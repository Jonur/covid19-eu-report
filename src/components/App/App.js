import React from 'react';
import { APP_DATA } from '../../definitions/propTypes';
import { Footer, GraphSection, Header, TableStatSection } from '..';
import {
  getCountiesTotalsDate,
  getEUTotalsByDateNewestFirst,
} from '../../utils/dataFilteringUtils';
import { formatThousands, getEUtotals } from '../../utils/graphUtils';
import s from './App.module.scss';

const App = ({ euCovidData, lastUpdate }) => {
  const countiesTotalsToDate = getCountiesTotalsDate(euCovidData);
  const euTotalsByDate = getEUTotalsByDateNewestFirst(euCovidData);
  const totals = getEUtotals(euTotalsByDate);

  return (
    <>
      <Header lastUpdate={lastUpdate} />
      <main aria-label="EU COVID-19 Charts" className={s.statWrapper}>
        <TableStatSection
          alerting="danger"
          ariaLabelledBy="total-deaths-title"
          countryStatColumnName="Deaths"
          data={countiesTotalsToDate}
          dataProp="totalDeaths"
          dataPropSecondary="deathsLast24h"
          sectionTitle={`${formatThousands(totals.deaths)} Total deaths`}
          sectionSubtitle="(+ COVID-19 caused deaths in the past 24 hours)"
          title="Total deaths table"
        />
        <TableStatSection
          alerting="warning"
          ariaLabelledBy="total-cases-title"
          countryStatColumnName="Cases"
          data={countiesTotalsToDate}
          dataProp="totalCases"
          dataPropSecondary="casesLast24h"
          sectionTitle={`${formatThousands(
            totals.confirmed
          )} Total confirmed cases`}
          sectionSubtitle="(+ COVID-19 confirmed cases in the past 24 hours)"
          title="Total confirmed cases table"
        />
        <TableStatSection
          alerting="success"
          ariaLabelledBy="total-recovered-title"
          countryStatColumnName="Recovered"
          data={countiesTotalsToDate}
          dataProp="totalRecovered"
          dataPropSecondary="recoveredLast24h"
          sectionTitle={`${formatThousands(
            totals.recovered
          )} Patients recovered`}
          sectionSubtitle="(+ COVID-19 recovered patients in the past 24 hours)"
          title="Recovered patients table"
        />
        <GraphSection
          alerting="success"
          ariaLabelledBy="increase-timeline-title"
          graphData={euTotalsByDate}
          sectionTitle="EU Incidents Timeline"
          sectionSubtitle="COVID-19 timeline in the European Union"
          totals={totals}
        />
      </main>
      <Footer />
    </>
  );
};

App.propTypes = APP_DATA;

export default App;
