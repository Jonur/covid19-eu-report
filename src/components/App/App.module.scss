.appLoading {
  align-self: center;
}

.statWrapper {
  flex: 1;
  padding: 0 0.5rem;
  margin: 1.75rem 0 0;
}
