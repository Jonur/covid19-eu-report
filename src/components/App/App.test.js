import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('App', () => {
  const props = {
    euCovidData: {},
    lastUpdate: '23 May 2020',
  };

  test('renders the App component correctly', () => {
    const { container } = render(<App {...props} />);
    expect(container).toMatchSnapshot();
  });
});
