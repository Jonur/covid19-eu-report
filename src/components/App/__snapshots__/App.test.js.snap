// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the App component correctly 1`] = `
<div>
  <header
    class="appHeader"
  >
    <img
      alt="EU flag"
      src="european-union.png"
      title="EU flag"
    />
    <h1
      class="appTitle"
    >
      EU COVID-19 Country Statistics
    </h1>
    <p
      class="lastUpdate"
    >
      Data collected on 23 May 2020
    </p>
  </header>
  <main
    aria-label="EU COVID-19 Charts"
    class="statWrapper"
  >
    <section
      aria-labelledby="total-deaths-title"
      class="section"
    >
      <h2
        class="sectionTitle"
        id="total-deaths-title"
      >
        Total deaths
        <span
          class="last24hInfo"
        >
          (+ COVID-19 caused deaths in the past 24 hours)
        </span>
        <span
          class="expandCollapseArrow"
        />
      </h2>
    </section>
    <section
      aria-labelledby="total-cases-title"
      class="section"
    >
      <h2
        class="sectionTitle"
        id="total-cases-title"
      >
        Total cases
        <span
          class="last24hInfo"
        >
          (+ COVID-19 confirmed cases in the past 24 hours)
        </span>
        <span
          class="expandCollapseArrow"
        />
      </h2>
    </section>
    <section
      aria-labelledby="total-recovered-title"
      class="section"
    >
      <h2
        class="sectionTitle"
        id="total-recovered-title"
      >
        Patients recovered
        <span
          class="last24hInfo positive"
        >
          (+ COVID-19 recovered patients in the past 24 hours)
        </span>
        <span
          class="expandCollapseArrow"
        />
      </h2>
    </section>
  </main>
  <footer
    class="globalFooter"
  >
    Â©
    2020 - 
    <a
      alt="Jonur on Github"
      class="link"
      href="https://github.com/Jonur"
      rel="noopener noreferrer"
      target="_blank"
    >
      Jonur
    </a>
     - 
    <a
      alt="Application Privacy Policy"
      class="link"
      href="https://github.com/Jonur/covid19-eu-report/blob/master/PrivacyPolicy.md"
      rel="noopener noreferrer"
      target="_blank"
    >
      Privacy Policy
    </a>
  </footer>
</div>
`;
